metabase/metabase
# 29553 add actions button is shown even when actions are not enabled for the database

Closes https://github.com/metabase/metabase/issues/29553

### Description

We want to remove the `Add Actions` button from dashboards whose data sources are databases that don't allow model actions. This change will only apply to dashboards where _all_ cards use data from those databases. Otherwise, things will remain as they are now.

### How to verify

1. Go to Metabase Admin, and ensure that the "Enable Model Actions" toggle is disabled within the Sample Database
2. Go to an existing dashboard whose cards only use Sample data and see that the "Add actions" button isn't there.
3. Go to an existing dashboard whose cards have different sources (where cards source data from at least one database with model actions) and see that the "Add actions" button is still there.
4. You can also go back to Metabase Admin, re-enable the model actions in the sample database, and go back to the dashboard from step (2) and see that the "Add actions" button is back.

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29863)
